app/views/ShareListView/index.js

```diff
@@ -0,0 +1,446 @@
+					type: mime.lookup(data.path),
+					store: 'Uploads',
+					path: isIOS ? data.path : `file://${ data.path }`
+				};
+			}
+			this.setState({
+				value, fileInfo, isMedia, mediaLoading: false
+			});
+		} catch (e) {
+			log('err_process_media_share_extension', e);
+			this.setState({ mediaLoading: false });
+		}
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const { loadingServer, searchText } = this.props;
+
+		if (nextProps.server && loadingServer !== nextProps.loadingServer) {
+			if (nextProps.loadingServer) {
+				this.internalSetState({ loading: true });
+			} else {
+				this.getSubscriptions();
+			}
+		} else if (searchText !== nextProps.searchText) {
+			this.search(nextProps.searchText);
+		}
+	}
+
+	shouldComponentUpdate(nextProps, nextState) {
+		const { loading, searching } = this.state;
+		if (nextState.loading !== loading) {
+			return true;
+		}
+		if (nextState.searching !== searching) {
+			return true;
+		}
+
+		const { search } = this.state;
+		if (!isEqual(nextState.search, search)) {
+			return true;
+		}
+		return false;
+	}
+
+	// eslint-disable-next-line react/sort-comp
+	internalSetState = (...args) => {
+		const { navigation } = this.props;
+		if (isIOS && navigation.isFocused()) {
+			LayoutAnimation.easeInEaseOut();
+		}
+		this.setState(...args);
+	}
+
+	getSubscriptions = debounce(() => {
+		if (this.data && this.data.removeAllListeners) {
+			this.data.removeAllListeners();
+		}
+
+		const { serversDB } = database.databases;
+		const { server } = this.props;
+
+		if (server && this.hasActiveDB()) {
+			this.data = database.objects('subscriptions').filtered('archived != true && open == true');
+			this.discussions = this.data.filtered('prid != null');
+			this.channels = this.data.filtered('t == $0 AND prid == null', 'c');
+			this.privateGroup = this.data.filtered('t == $0 AND prid == null', 'p');
+			this.direct = this.data.filtered('t == $0 AND prid == null', 'd');
+			this.livechat = this.data.filtered('t == $0 AND prid == null', 'l');
+			this.servers = serversDB.objects('servers');
+			safeAddListener(this.data, this.updateState);
+		}
+	}, 300);
+
+	uriToPath = uri => decodeURIComponent(isIOS ? uri.replace(/^file:\/\//, '') : uri);
+
+	// eslint-disable-next-line react/sort-comp
+	updateState = debounce(() => {
+		this.updateStateInteraction = InteractionManager.runAfterInteractions(() => {
+			this.internalSetState({
+				discussions: this.discussions ? this.discussions.slice() : [],
+				channels: this.channels ? this.channels.slice() : [],
+				privateGroup: this.privateGroup ? this.privateGroup.slice() : [],
+				direct: this.direct ? this.direct.slice() : [],
+				livechat: this.livechat ? this.livechat.slice() : [],
+				servers: this.servers ? this.servers.slice() : [],
+				loading: false
+			});
+			this.forceUpdate();
+		});
+	}, 300);
+
+	// this is necessary during development (enables Cmd + r)
+	hasActiveDB = () => database && database.databases && database.databases.activeDB;
```

Comment:
I think you can delete this line here and the calls.
You can test really... remove it and refresh simulator.
It's necessary only during development in some cases due to a Realm bug IIRC.